apps/astarte_realm_management/lib/astarte_realm_management/engine.ex

```diff
@@ -457,18 +459,36 @@ defmodule Astarte.RealmManagement.Engine do
          trigger_target = target_from_action(action_map, trigger_uuid),
          t_container = build_trigger_target_container(trigger_target),
          :ok <- validate_simple_triggers(realm_name, simple_trigger_maps),
-         # TODO: these should be batched together
          :ok <-
            install_simple_triggers(realm_name, simple_trigger_maps, trigger_uuid, t_container),
-         :ok <-
-           install_trigger_policy_link(realm_name, trigger_uuid, trigger_policy_name) do
+         :ok <- install_trigger_policy_link(realm_name, trigger_uuid, trigger_policy_name),
+         result <- Queries.install_trigger(realm_name, trigger) do
       _ =
-        Logger.info("Installing trigger.",
+        Logger.info("Trigger installation completed successfully.",
           trigger_name: trigger_name,
-          tag: "install_trigger_started"
+          tag: "install_trigger_completed"
         )
 
-      Queries.install_trigger(realm_name, trigger)
+      extracted_simple_trigger = hd(Enum.map(simple_trigger_maps, & &1.simple_trigger))
+
+      simple_trigger_data = extracted_simple_trigger.simple_trigger
+
+      request_data = %{
+        object_id: hd(Enum.map(simple_trigger_maps, & &1.object_id)),
+        object_type: hd(Enum.map(simple_trigger_maps, & &1.object_type)),
+        parent_trigger_id: trigger_target.parent_trigger_id,
+        simple_trigger_id: hd(Enum.map(simple_trigger_maps, & &1.simple_trigger_uuid)),
+        simple_trigger: simple_trigger_data,
+        trigger_target: trigger_target
+      }
+
+      Logger.info(
+        "Sending trigger installation notification to DataUpdaterPlant for trigger #{inspect(request_data.simple_trigger)} ..."
+      )
+
+      Client.install_persistent_trigger(request_data)
```

Comment:
This call the actual Client.
I saw you prepared the scaffolding to mock the behavior in tests with Mox, and so the module to use here should be `Astarte.RealmManagement.RPC.DataUpdaterPlant.Trigger` (`apps/astarte_realm_management/lib/astarte_realm_management/rpc/data_updater_plant.ex`).

Alternatively, you can follow @lusergit good suggestion to avoid that scaffolding, just have and use the actual Client module implementation. During tests, we should be able to use [Mimic](https://github.com/edgurgel/mimic) to mock the client's functionality and control it as we wish (without the behavior + mocks that one would usually have for working with the [Mox](https://hexdocs.pm/mox/Mox.html) library).